app/jobs/lti_roster_sync_job.rb

```diff
@@ -9,9 +9,13 @@ def self.completed_message(_status)
     I18n.t('lti.roster_sync_complete')
   end
 
-  def perform(lti_deployment, course, role_types, can_create_users: false, can_create_roles: false)
-    roster_error = roster_sync(lti_deployment, course, role_types, can_create_users: can_create_users,
-                                                                   can_create_roles: can_create_roles)
+  def perform(args)
+    args = args.deep_symbolize_keys
+    lti_deployment = LtiDeployment.find(args[:deployment])
+    course = lti_deployment.course
```

Comment:
It seems like defining a `course` variable is redundant here; can you modify `roster_sync` so that it doesn't take in a `course` argument?